staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -945,6 +959,19 @@ func (alloc *allocator) allocateDevice(r deviceIndices, device deviceWithID, mus
 		return false, nil, nil
 	}
 
+	if alloc.features.PartitionableDevices && len(device.slice.Spec.CapacityPools) > 0 && len(device.basic.ConsumesCapacity) > 0 {
```

Comment:
Shouldn't this be:

```go
if alloc.features.PartitionableDevices && len(device.basic.ConsumesCapacity) > 0 {
  if len(device.slice.Spec.CapacityPools) == 0 {
    return false, nil, nil
  }
...check capacity...
}
```

Otherwise, if a device consumes capacity but there is none defined, we will allocate it. I guess if we add the validation of "consumes only capacities named in pools" then this will never happen.

Reply:
The validation should prevent this case(having consumesCapacity set without capacityPools) from happening so I combined the check :)

Reply:
maybe just a comment so people don't  wonder